examples/capture_circular_nooutput.py

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/python3
+import sys
+import time
+
+from picamera2 import Picamera2
+from picamera2.encoders import H264Encoder
+from picamera2.outputs import CircularOutput
+
+picam2 = Picamera2()
+fps = 30
+dur = 5
+micro = int((1 / fps) * 1000000)
+vconfig = picam2.create_video_configuration()
+vconfig['controls']['FrameDurationLimits'] = (micro, micro)
+picam2.configure(vconfig)
+encoder = H264Encoder()
+output = CircularOutput(buffersize=fps * dur, outputtofile=False)
+output.fileoutput = "file.h264"
+picam2.start_recording(encoder, output)
+time.sleep(dur)
```

Comment:
So this is where we don't seem to get a 5 second output, right? One thing to try might be to add
```
picam2.capture_metadata()
```
_before_ the `sleep` command (but after starting the recording). That will force it to wait for the first frame to come out of the camera before starting the timer. That will factor out any "startup time".

Reply:
Yep, that's correct.  Will try that approach, thanks.